src/libcsg/modules/io/pdbreader.cc

```diff
@@ -27,25 +29,32 @@ using namespace std;
 
 bool PDBReader::ReadTopology(string file,  Topology &top)
 {
-   _topology = true;
-   top.Cleanup();
+    _topology = true;
+    top.Cleanup();
 
-   _fl.open(file.c_str());
+    _fl.open(file.c_str());
     if(!_fl.is_open())
         throw std::ios_base::failure("Error on open topology file: " + file);
 
-   NextFrame(top);
-
+    NextFrame(top);
     _fl.close();
 
     return true;
 }
 
-bool PDBReader::Open(const string &file)
-{
+bool PDBReader::Open(const string &file){
+
+    boost::filesystem::path filepath(file.c_str());
+
+    // Check that the file has the correct extension
+    if(boost::filesystem::extension(filepath).size()==0 ){
+        throw std::ios_base::failure("ERROR on opening .pdb file'"
+                +file+"' - extension is expected, use .pdb");
+    }
```

Comment:
I think that is not needed, as `TrjReaderFactory()` already checks the extension and only call the pdb reader if it is a `pdb` file.

Reply:
Drop this!